File: client/src/app/modules/miscellaneous/services/country.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { Country } from '../../miscellaneous/model/Country';

@Injectable({
  providedIn: 'root'
})
export class CountryService extends AppRestService<Country>{

  constructor(http: HttpClient) {
    super(http, "miscellaneous");
  }

  getCountries() {
    return this.getListCached(new HttpParams(), "countries");
  }

  addOrUpdateCountry(country: Country) {
    this.clearListCache(new HttpParams(), "countries");
    return this.addOrUpdate(new HttpParams(), "country", country, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/miscellaneous/service/CountryService.java
package com.jss.osiris.modules.osiris.miscellaneous.service;

import java.util.List;

import com.jss.osiris.modules.osiris.miscellaneous.model.Country;

public interface CountryService {
    public List<Country> getCountries();

    public Country getCountry(Integer id);

    public Country addOrUpdateCountry(Country country);
}
